### ISSUE TITLE
Depend on directories in repo rules (recursively)

### ISSUE THREAD

**Issue by matts1** [bazelbuild/bazel#19928]
### Description of the feature request:

We have two repo rules internally that depend on whole directories.

We have a rust binary that needs to generate a repository. Because of that, it can't depend on a toolchain, and we have to it in a repo rule.

In order to build the rust binary, we need our repo rule to depend on.

```
build_rust_binary = rule(
    attrs = dict(srcs = attr.label_list(...))
)
```

```
src/main.rs
src/foo/bar.rs
...
```

At the moment, we need to write `build_rust_binary(srcs=["src/main.rs", "src/foo/bar.rs", ...])`. However, that means that if we add a new file `src/foo/baz.rs`, for example, it won't trigger a rebuild, and it may or may not actually build successfully, depending on the implementation.

We have similar issues with another repo rule that depends on a directory, where each directory in the input corresponds to a directory in the output.

I'd like to be able to specify a directory as a label for inputs, so any changes to the transitive closure of the direcotry will trigger a rebuild.

### Which category does this issue belong to?

External Dependency

### What underlying problem are you trying to solve with this feature?

Ensure that rebuilds are triggered when a directory is modified (or any subdirectory)

### Which operating system are you running Bazel on?

Linux

### What is the output of `bazel info release`?

6.2.0

**Comment by meteorcloudy** [comment]
https://docs.google.com/document/d/17RZKMuMjAIgNfFdrsLhqNsMTQDSS-BBA-S-fCSBPV7M/edit should help

**Comment by Wyverald** [comment]
dupe of https://github.com/bazelbuild/bazel/issues/20952 (which is fixed and will be in 7.1.0)
